### DISCUSSION TITLE
How to insert into storage.objects using SQL

### DISCUSSION THREAD

**Discussion by LianBond** [supabase/supabase#12694] [Questions] [answered]
Hi, 

I want to use SQL to insert an image into a bucket called 'images'.

This is the current insert statement I have 
`INSERT INTO storage.objects (bucket_id, name, data) VALUES ('images', new_image_name, file_data);`

but I'm getting column \"data\" of relation \"objects\" does not exist. 

How can I How to insert into storage.objects using SQL?

**Comment by GaryAustin1** [comment, marked as answer]
You have to make storage-api calls to upload files.  You should never modify the storage.objects table directly.  That is just file info for Supabase to keep track of the actual files stored in the s3 bucket.

I have done and seen code to delete files with the http extension making a REST call to the Storage API, but I don't think I've seen anyone writing a file from within the database.  Not saying it can't be done, but it will involve making a REST call to the Storage API.

**Reply by GaryAustin1** [reply]
Many of the guides now show the SQL code to delete a file from the SQL side, this gives you an idea of what is involved.
https://supabase.com/docs/guides/getting-started/tutorials/with-nextjs#storage-management

**Reply by LianBond** [reply]
Gary,

Thanks so much for the prompt response. I want to do this because I want to auto-increment the file name when there are duplicates in a storage bucket.

You don't recommend modifying the storage.objects table directly, which makes sense. Is there any other way you would recommend me doing this?

**Reply by GaryAustin1** [reply]
You could possibly use an edge function called from a trigger, if that is how you were wanting to insert to begin with, you don't really say where your insert call was going to be or how it got file data.

You can search and/or figure out how to use the http extension (like the link I showed for deleting a file) and use that to make a storage-api upload call.

**Reply by LianBond** [reply]
So this is for when the user uploads an image on the client side. Then the image must be sent to supabase and autoincremented in case of duplicate file names. I think you are right about edge functions, was not aware of this, so thank you for pointing me in that direction. Will try this and let you know. Thanks again!

**Reply by GaryAustin1** [reply]
Do you really care what the file name is?  You might just embed a timestamp or UUID as part of the pathname then don't have to worry about it.   Like  images/myfilename-[insert UUID here].png or images/UUID-filename.png.

**Comment by LianBond** [comment]
Not sure why I didn’t think of that. Silly me. I’ll do both the uuid as
well as the timestamp. That should always be unique. Thanks sir!
